rust/connlib/clients/shared/src/eventloop.rs

```diff
@@ -74,6 +85,28 @@ where
                 Poll::Pending => {}
             }
 
+            if let Poll::Ready(Some(dns)) = self.dns_updated.poll_recv(cx) {
+                self.dns_update = Some(
+                    async move {
+                        // We sleep here to debounce since if another event is generated within the given time frame it will replace this future.
+                        tokio::time::sleep(Duration::from_millis(100)).await;
+                        dns
+                    }
+                    .boxed(),
+                );
+            }
+
+            if let Some(mut dns_update) = self.dns_update.take() {
+                match dns_update.poll_unpin(cx) {
+                    Poll::Ready(dns) => {
+                        self.tunnel.set_dns(dns);
```

Comment:
That will allows us to unit-test the debouncing and it doesn't require messing around with futures. All we need is an `Option` with an `Instant` and act on it in `handle_timeout`.

Reply:
the downside with this approach is that `handle_timeout` would need to call into `Tunnel` to call `update_interface`, so we would need to return some event from `handle_timeout`, if that's acceptable I almost have an implementation for this.

Reply:
`handle_timeout` shouldn't return anything, instead we should queue an event of some kind that is then consumed by the `Tunnel` so it can update its IO state.